### ISSUE TITLE
`no-return-assign` behavior changed with arrow functions (again ?)

### ISSUE THREAD

**Issue by Kerumen** [eslint/eslint#9359]
https://github.com/eslint/eslint/issues/5150 seems to resurface.

**Tell us about your environment**

* **ESLint Version:** `v4.7.2`
* **Node Version:** `v8.5.0`
* **npm Version:** `v.5.4.2`

**What parser (default, Babel-ESLint, etc.) are you using?**

`babel-eslint`

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "extends": "airbnb",
  "parser": "babel-eslint",
  "env": {
    "browser": true,
    "node": true,
    "es6": true
  },
  "globals": {
    "$env": false,
    "window": true,
    "__DEV__" : false,
    "__PROD__" : false,
    "__DEBUG__" : false,
    "__COVERAGE__" : false,
    "__BASENAME__" : false
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue.**

```js
<div ref={c => (this.container = c)}></div>
```

**What did you expect to happen?**

In previous version of ESlint, the `no-return-assign` error could be fixed by adding parenthesis around the assignment. This doesn't work anymore, and it's particularly annoying for React's ref.

**What actually happened? Please include the actual, raw output from ESLint.**
```
error    Arrow function should not return assignment  no-return-assign
```

**Comment by platinumazure** [comment]
In normal JS, the correct thing to do would be to use a block statement:

```js
c => { this.container = c; }
```

But I assume this won't work in JSX?

**Comment by not-an-aardvark** [comment]
I can't reproduce this issue in the demo (see [here](https://eslint.org/demo/#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXJldHVybi1hc3NpZ246IGVycm9yICovXG5cbjxkaXYgcmVmPXtjID0+ICh0aGlzLmNvbnRhaW5lciA9IGMpfT48L2Rpdj4iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjYsInNvdXJjZVR5cGUiOiJtb2R1bGUiLCJlY21hRmVhdHVyZXMiOnsianN4Ijp0cnVlfX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19)).

**Comment by Kerumen** [comment]
Ok it's totally my bad here. Airbnb updated his config (in this commit: https://github.com/airbnb/javascript/commit/57ff032b0740ba2a46af4bc40cf5638a3e62a365) by strengthening the rule with the `always` param. I wasn't aware of this param so I thought eslint was reporting a false positive.

Sorry!
